browser/profiles/brave_profile_impl.cc

```diff
@@ -39,9 +39,8 @@ BraveProfileImpl::BraveProfileImpl(
   if (brave::IsSessionProfilePath(path) &&
       create_mode == CREATE_MODE_ASYNCHRONOUS) {
     auto* parent_profile = brave::CreateParentProfileData(this);
+    parent_profile->AddObserver(this);
```

Comment:
we'd better unsubscribe manually, or use `ScopedObserver`

Reply:
Fixed in eb44f100920c0456ea21618a716fe2af70938356